src/settings.ts

```diff
@@ -8,6 +8,17 @@ const airgapInit = (self as any)?.airgap;
 // eslint-disable-next-line @typescript-eslint/no-explicit-any, no-restricted-globals
 const transcendInit = (self as any)?.transcend;
 
+const { currentScript: thisScript } = document;
+
+export const scriptLocation = new URL(
```

Comment:
how well is this tested cross browser? im a bit scared to make this change because the URL constructor throws an error if passed invalid arguments

Reply:
It's more a matter of testing across different embed environments as opposed to cross-browser, as all browsers behave the same with this statement.

If `document.currentScript` is null, then `scriptLocation` defaults to `location.href`. If `document.currentScript` is an HTML script & it has an src attribute, then that attribute is used for the location. Similarly, if it's an SVG script we extract the appropriate attribute in that case as well.

In any case, if a proper value cannot be determined we fall back to the always-valid `location.href`.

Reply:
ok, just saying, im not sure we need this, and the code here scares me because it could 100% throw an error if we have a mistake. and given i dont think we need this, i would do a ton of testing before we merge this in